### PR TITLE
Handle 'tab' and 'enter' key press in file search input field

### PR DESCRIPTION
**Pull request by kareniel**
Associated Issue: #971 

### Summary of Changes

This commit adds conditions that handle the cases in which
a user presses the 'tab' or the 'enter' key while the file search
input field (autocomplete component) is focused.

- modify case for 'enter' key in Autocomplete's onKeyDown method
- add case for 'tab' key in Autocomplete's onKeyDown() method
- add previousInput:string field to 'ui' state store
- add getFileSearchPreviousInput getter to 'ui' state store
- add saveFileSearchInput action creator in 'ui' actions
- add corresponding constant
- add case for that constant in 'ui' reducers
- pass saveFileSearchInput to SourceSearch's props
- write test for saveFileSearchInput action creator
- set cursor at the end of the input field's content

### Testing

* [x] passes `npm test`
* [x] passes `npm run lint` ( this commit doesn't add any additional rule breaks )

### CODE REVIEW

**Comment by jasonLaster** [comment]
Thanks @kareniel. This was a fantastic first PR. 

One question, what are the trade-offs between having the search input in the redux store or source search state? Are there advantages to having it in the store that you wouldn't get in component state?

Also, noticed that there is one conflict. Could you quickly rebase your branch off of an uptodate master?

**Comment by kareniel** [comment]
@jasonLaster : Actually, that piece of state could probably live in the `source search` component.  I only see a case for this if we think that component will get destroyed. I assumed this was the case, because of the existence of `searchOn` (and it being the only variable stored in the `ui` store).

**Comment by jasonLaster** [comment]
> that piece of state could probably live in the source search

Yup, I agree. We added `searchOn` to the UI state because the "+" button in the source tabs row needs to open source search with a redux action. We're not worried about the component unmounting.

> I updated the history 

that's fine. I can squash these commits when I merge it in.

**Comment by kareniel** [comment]
Gotcha. I can probably get this done later today.

File: public/js/reducers/ui.js [hunk 1]
```diff
@@ -12,18 +12,23 @@ import type { Action } from "../actions/types";
...
+  searchOn: false,
+  previousInput: ""
 } : UIState));
 
 function update(state = State(), action: Action): Record<UIState> {
   switch (action.type) {
     case constants.TOGGLE_FILE_SEARCH: {
       return state.set("searchOn", action.searchOn);
     }
+    case constants.SAVE_FILE_SEARCH_INPUT: {
```
Comment (jasonLaster) [inline public/js/reducers/ui.js]: oops, this stuff needs to go

File: public/js/components/SourceSearch.js [hunk 2]
```diff
@@ -83,17 +91,20 @@ const Search = React.createClass({
       Autocomplete({
         selectItem: result => {
           this.props.selectSource(result.id);
+          this.setState({ previousInput: "" });
           this.props.toggleFileSearch(false);
         },
         handleClose: this.close,
-        items: searchResults(this.props.sources)
+        items: searchResults(this.props.sources),
+        previousInput: this.state.previousInput
```
Comment (jasonLaster) [inline public/js/components/SourceSearch.js]: could we call this `searchValue` or `inputValue`? Sometimes with this stuff, i wonder if we need the tense for instance does "previous" make it more clear or is "current" redundant... The answer could well be, previous helps for X reason, i just haven't thought about it much.

**Pull request merged**